Add cards and badges

TotomInc opened the pull request:
Closes #11 to put cards and badges together. WIP on modern web components as #3 says.

Fraham commented on src/badges.less:
```diff
@@ -0,0 +1,35 @@
+  text-align: center;
+  white-space: nowrap;
+  vertical-align: baseline;
+  background-color: #868e96;
```
Shouldnt this be assigned a colour variable?

TotomInc replied:
Yeah totally forgot about that, I'll replace it asap!

rhyneav commented on src/cards.less:
```diff
@@ -0,0 +1,55 @@
+  // since we want to avoid the little blank space between
+  // image and border-top-left/right-radius, so we force
+  // the default border class and customize it on the img
+  .border;
```
Maybe for cards, we don't need to use the crooked borders like shown in the other elements. It could be shown as something similar to a "Note Card" which has straight lines. That way the button within it stands out more. And then we don't have to worry about adjusting the image style from the border effects.

![image](https://user-images.githubusercontent.com/13633775/32299926-b74e6c80-bf2d-11e7-92a5-b3cae8692c9c.png)
![image](https://user-images.githubusercontent.com/13633775/32299994-fc69366a-bf2d-11e7-8991-356370f89b32.png) 

But I like the direction so far!

TotomInc replied:
Sure, I'll fix it asap. I have to admit that the crooked borders was not very good for the cards. I'll fix that and add a box shadow like on the pic, and also add header/footer customization on cards.

rhyneav commented:
Couple comments above for the cards (and badge color variable). But I also wanted to say that the badges look sharp. Great stuff!

![image](https://user-images.githubusercontent.com/13633775/32300076-4fa2948e-bf2e-11e7-99cb-934c55ff6106.png)

TotomInc commented:
I really like how the badges are and follow properly the paper theme, but if you feel they are too sharp, should I just put another border radius? How do you see the badges?

TotomInc commented:
I've just fixed the hard-coded color on badges and added a new one, that I think will be used especially for text (.muted, a gray and its lighten version). I also removed *edgy* corners on the badges, don't know if that what @rhyneav wanted. I can still change it depending on what you want.

rhyneav commented:
@TotomInc sorry, I meant that they look sharp as in James Bond in a tuxedo type of sharp. I really liked how they matched the theme with the uneven borders rather than the rounded radius. 

What I'm trying to say is that your original badges looked great as this:
![image](https://user-images.githubusercontent.com/13633775/32330695-3f1c7650-bfb7-11e7-9ede-3cd14628f98d.png)

TotomInc commented:
Oops, sorry I misunderstand your comment on badges, I'm going to revert the changes! Also, did you check the new cards? I removed the paper-style borders and added a shadow.

TotomInc commented:
The PR is ready to be merged @rhyneav .

Fraham commented:
Looks really good

TotomInc commented:
Glad you like it! 😄

rhyneav commented:
Wow, cards and badges both look stellar! Thank you for doing such an awesome job with those. I also really like the addition of the muted color; much needed!
